src/pynwb/image.py

```diff
@@ -73,19 +79,104 @@ def __init__(self, **kwargs):
         if unit is None:
             kwargs['unit'] = ImageSeries.DEFAULT_UNIT
 
+        if (
+            args_to_set["external_file"] is not None
+            and args_to_set["starting_frame"] is None
+        ):
+            args_to_set["starting_frame"] = (
+                [0] if len(args_to_set["external_file"]) == 1 else None
+            )
+
         super().__init__(**kwargs)
 
-        if args_to_set["external_file"] is None:
-            args_to_set["starting_frame"] = None  # overwrite starting_frame
         for key, val in args_to_set.items():
             setattr(self, key, val)
 
+        if not self._check_external_file_starting_frame_length():
+            msg = ("%s '%s': The number of frame indices in 'starting_frame' should have "
+                   "the same length as 'external_file'." % (self.__class__.__name__, name))
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if self._change_external_file_format():
+            warnings.warn(
+                "%s '%s': The value for 'format' has been changed to 'external'. "
+                "Setting a default value for 'format' is deprecated and will be changed "
+                "to raising a ValueError in the next release."
+                % (self.__class__.__name__, self.name),
+                DeprecationWarning,
+            )
+
+        if not self._check_external_file_format():
+            msg = (
+                "%s '%s': Format must be 'external' when external_file is specified."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if not self._check_external_file_data():
+            msg = (
+                "%s '%s': Either external_file or data must be specified (not None), but not both."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
         if not self._check_image_series_dimension():
             warnings.warn(
                 "%s '%s': Length of data does not match length of timestamps. Your data may be transposed. "
-                "Time should be on the 0th dimension" % (self.__class__.__name__, self.name)
+                "Time should be on the 0th dimension"
+                % (self.__class__.__name__, self.name)
             )
 
+    def _raise_error_when_check_is_violated(self, error_msg):
+        """Raise an error when a check is violated on instance creation.
+        To ensure backwards compatibility, this method throws a warning
+        instead of raising an error when reading from a file."""
+        if not self._in_construct_mode:
+            raise ValueError(error_msg)
+        warnings.warn(error_msg)
```

Comment:
@oruebel let me know what you think about this, I added tests in [tests/back_compat/test_read.py](https://github.com/NeurodataWithoutBorders/pynwb/pull/1516/files#diff-7f35c9a018fa89011cfaf801cfcc8747edbeefc94ab48e6ab0862469407f57e2) to see if warnings are raised. I'm not sure how to fix coverage for L136, I thought the back compat tests are sufficient for testing the warnings.